todo_interface.rb

```diff
@@ -0,0 +1,165 @@
+require_relative 'todo_item_class.rb'
+require_relative 'todo_list_class.rb'
+require 'colorize'
+require 'date'
+
+
+def input_date
+	while(true)
```

Comment:
Ruby `while` loops tend to have a specific end condition. Instead of looping while true, you could, for example, loop until you have a proper date to return. For example:
```
date = nil
until date
  puts "Now enter the due date in the form \"dd-mm-yyyy\": "
  date_input = gets.chomp
  d, m, y = date_input.split('-')
  if Date.valid_date? y.to_i, m.to_i, d.to_i
    begin
      date = Date.parse(date_input)
    rescue Exception => e
    end
  else
      puts "Oops! Invalid date -- try again!".red
  end
end
date
```

This will run until date is a truthy value, and then implicitly return date at the end of the method.

Also, you're rescuing an error, but not doing anything with it. Is that left over from before you were checking to see if the date was valid?